stingray/lightcurve.py

```diff
@@ -451,8 +451,8 @@ def join(self, other):
             except IndexError:
                 count2 = None
 
-            if not count1 is None:
-                if not count2 is None:
+            if count1 is not None:
+                if count2 is not None:
```

Comment:
No, let's leave the explicit check; I agree with you on that. I was mostly curious. :)